File: app/src/main/java/io/zenandroid/greenfield/base/BasePresenter.java
package io.zenandroid.greenfield.base;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * Created by acristescu on 02/07/2017.
 */

public abstract class BasePresenter implements Presenter {

	private View view;
	private CompositeDisposable compositeDisposable = new CompositeDisposable();

	public BasePresenter(View view) {
		this.view = view;
	}

	protected void onError(Throwable throwable) {
		view.dismissProgressDialog();
		view.showErrorMessage(throwable.getMessage());
	}

	protected void addDisposable(Disposable disposable) {
		compositeDisposable.add(disposable);
	}

	@Override
	public void unsubscribe() {
		compositeDisposable.clear();
	}
}

File: app/src/main/java/io/zenandroid/greenfield/base/BasePresenter.kt
package io.zenandroid.greenfield.base

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

/**
 * Created by acristescu on 02/07/2017.
 */

abstract class BasePresenter(private val view: View<*>) : Presenter {
    private val compositeDisposable = CompositeDisposable()

    protected fun onError(throwable: Throwable) {
        view.dismissProgressDialog()
        view.showErrorMessage(throwable.message)
    }

    protected fun addDisposable(disposable: Disposable) {
        compositeDisposable.add(disposable)
    }

    override fun unsubscribe() {
        compositeDisposable.clear()
    }
}
